File: Submissions/821_Shortest_Distance_to_a_Character.cpp
class Solution {
public:
    vector<int> shortestToChar(string S, char C) {
        int n = S.size();
        vector<int> result(n, n);
        int prev = -n;

        for(int i=0; i<n; ++i){
            if(S[i] == C) prev = i;
            result[i] = i-prev;
        }
        
        prev = 2*n;
        for(int i=n-1; i>=0; --i){
            if(S[i] == C) prev = i;
            result[i] = min(result[i], prev-i);
        }
        
        return result;
    }
};

File: Kotlin/821_Shortest_Distance_to_a_Character.kt
class Solution {
    fun shortestToChar(s: String, c: Char): IntArray {
        val n = s.length
        val result = IntArray(n){n}
        var prev = -n
        
        for(i in s.indices) {
            if(s[i] == c) prev = i
            result[i] = i - prev
        }
        
        prev = 2 * n
        for(i in s.indices.reversed()) {
            if(s[i] == c) prev = i
            result[i] = Math.min(result[i], prev - i)
        }
        
        return result
    }
}